tensorboard/data/server/data_compat.rs

```diff
@@ -619,6 +650,76 @@ mod tests {
         }
     }
 
+    mod tensors {
+        use super::*;
+
+        #[test]
+        fn test_metadata_tensor_with_dataclass() {
+            let md = pb::SummaryMetadata {
+                plugin_data: Some(PluginData {
+                    plugin_name: "rando".to_string(),
+                    content: Bytes::from_static(b"preserved!"),
+                    ..Default::default()
+                }),
+                data_class: pb::DataClass::Tensor.into(),
+                ..Default::default()
+            };
+            let v = SummaryValue(Box::new(Value::Tensor(pb::TensorProto {
+                dtype: pb::DataType::DtString.into(),
+                string_val: vec![Bytes::from_static(b"foo")],
+                ..Default::default()
+            })));
+            let result = v.initial_metadata(Some(md.clone()));
+            assert_eq!(*result, md);
+        }
+
+        #[test]
+        fn test_metadata_tensor_without_dataclass() {
+            for plugin_name in vec![
```

Comment:
Clippy says:

```
warning: useless use of `vec!`
   --> data_compat.rs:678:32
    |
678 |               for plugin_name in vec![
    |  ________________________________^
679 | |                 plugin_names::HISTOGRAMS,
680 | |                 plugin_names::TEXT,
681 | |                 plugin_names::PR_CURVES,
...   |
684 | |                 plugin_names::NPMI,
685 | |             ] {
    | |_____________^
    |
    = note: `#[warn(clippy::useless_vec)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_vec
help: you can use a slice directly
    |
678 |             for plugin_name in &[plugin_names::HISTOGRAMS,
679 |                 plugin_names::TEXT,
680 |                 plugin_names::PR_CURVES,
681 |                 plugin_names::HPARAMS,
682 |                 plugin_names::MESH,
683 |                 plugin_names::NPMI] {
    |
```

(In case you’re wondering, the reason that these checks don’t show up
inline is that [our Cargo package is not at repository root][1].)

[1]: https://github.com/actions-rs/clippy-check/issues/28#issuecomment-680401392

Reply:
Done, thanks.